corehq/apps/hqwebapp/forms.py

```diff
@@ -21,6 +24,23 @@ def clean_username(self):
         username = self.cleaned_data['username'].lower()
         return username
 
+    def clean(self):
+        lockout_message = mark_safe(_('Sorry - you have attempted to login with an incorrect password too many times. Please <a href="/accounts/password_reset_email/">click here</a> to reset your password.'))
+        username = self.cleaned_data['username']
+        try:
+            cleaned_data = super(EmailAuthenticationForm, self).clean()
+        except ValidationError:
+            user = CouchUser.get_by_username(username)
+            if user and user.is_web_user() and user.is_locked_out():
+                raise ValidationError(lockout_message)
+            else:
+                raise
+        user = CouchUser.get_by_username(username)
+        if user and user.is_web_user() and user.is_locked_out():
+            raise ValidationError(lockout_message)
```

Comment:
Im sorry, I am not explaining this well. Everything you said is correct, we do still check credentials for locked out users but if they are locked out we return a `ValidationError` no matter what. The reason that code is there is for a user that has been wrong 4 times. On their 5th attempt, to login will either go through normally, or return the lockout message as opposed to the normal `wrong credentials` message. The call in the try block increments the attempts counter which is why the result of `user.is_locked_out()` may be different in the except block and before it. Otherwise, on the 5th failed attempt, a normal `wrong credentials` message is returned even though the user will be unable to login on the next try.